include/gdf/cffi/io_types.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2018, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * API
+ *
+ *
+ * gdf_column ** read_csv(csv_read_arg *args,  int *num_col_out, gdf_error *error_out);
+ *
+ */
+
+
+typedef struct {
+
+	/*
+	 * Output Arguments - space created in reader.
+	 */
+	const int		*num_cols;							// Out: number of columns
+	const int		*num_rows;							// Out: number of rows
```

Comment:
This would be tough to know ahead of time in many situations. We should likely try to address this via some kind of inferencing process in the future if possible.